pkg/scheduler/framework/types.go

```diff
@@ -80,9 +81,23 @@ type PodInfo struct {
 
 // AffinityTerm is a processed version of v1.PodAffinityTerm.
 type AffinityTerm struct {
-	Namespaces  sets.String
-	Selector    labels.Selector
-	TopologyKey string
+	Namespaces        sets.String
+	Selector          labels.Selector
+	TopologyKey       string
+	NamespaceSelector labels.Selector
+}
+
+// MergeNamespaces updates Namespaces with the set of namespaces identified by
+// NamespaceSelector.
+func (at *AffinityTerm) MergeNamespaces(nsLister listersv1.NamespaceLister) error {
+	ns, err := nsLister.List(at.NamespaceSelector)
+	if err != nil {
+		return err
+	}
+	for _, n := range ns {
+		at.Namespaces.Insert(n.Name)
+	}
+	return nil
```

Comment:
should we cleanup `NamespaceSelector`?

Reply:
good point, yes it is better to do that.